src/esp/sensor/CameraSensor.cpp

```diff
@@ -19,6 +19,17 @@ CameraSensor::CameraSensor(scene::SceneNode& cameraNode,
     : VisualSensor(cameraNode, spec),
       baseProjMatrix_(Magnum::Math::IdentityInit),
       zoomMatrix_(Magnum::Math::IdentityInit) {
+  // Sanity check
+  CORRADE_ASSERT(std::static_pointer_cast<VisualSensorSpec>(spec) != nullptr,
+                 "Spec given is of type VisualSensor", );
+  CORRADE_ASSERT(spec->sensorType == SensorType::Color,
+                 "Spec given has SensorType Color", );
+  CORRADE_ASSERT(std::static_pointer_cast<CameraSensorSpec>(spec) != nullptr,
+                 "Spec given is of type CameraSensor", );
+  CORRADE_ASSERT(
+      spec->sensorSubType == SensorSubType::Pinhole ||
+          spec->sensorSubType == SensorSubType::Orthographic,
+      "CameraSensorSpec given has SensorSubType Pinhole or Orthographic", );
```

Comment:
The error msg in the CORRADE_ASSERT always starts with the location, e.g., `CameraSensor::CameraSensor(): `
You need to modify your error msg so that it is informative when being displayed to the user (when an error occurs).
For example here:
`CameraSensor::CameraSensor(): Sensor sub-type" <<spec->sensorSubType <<"is not supported."`